core/src/commonMain/kotlin/com/xebia/functional/xef/auto/Conversation.kt

```diff
@@ -9,36 +9,44 @@ import com.xebia.functional.xef.llm.models.images.ImagesGenerationResponse
 import com.xebia.functional.xef.prompt.Prompt
 import com.xebia.functional.xef.vectorstores.ConversationId
 import com.xebia.functional.xef.vectorstores.VectorStore
-import kotlin.jvm.JvmName
-import kotlin.jvm.JvmOverloads
+import kotlin.jvm.JvmSynthetic
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.async
 import kotlinx.coroutines.flow.Flow
 import kotlinx.uuid.UUID
 import kotlinx.uuid.generateUUID
 
-/**
- * The [Conversation] is the context in which [AI] values are run. It encapsulates all the
- * dependencies required to run [AI] values, and provides convenient syntax for writing [AI] based
- * programs.
- */
-class Conversation
-@JvmOverloads
-constructor(
-  val store: VectorStore,
-  val conversationId: ConversationId? = ConversationId(UUID.generateUUID().toString())
-) : AutoCloseable, AutoClose by autoClose() {
+interface Conversation : AutoClose, AutoCloseable {
```

Comment:
This has changed to an interface to have the implementation be platform specific via `expect/actual` multiplatform declarations. This allows us to project only the `CompletableFuture` based methods over the JVM implementation.